rewriter/Flatten.cc

```diff
@@ -40,6 +40,17 @@ using namespace std;
 // has been replaced with the verbatim symbol `:bar`.
 namespace sorbet::rewriter {
 
+namespace {
+
+bool isAttrReaderMethodDef(core::Context ctx, ast::MethodDef &methodDef) {
+    auto *ivar = ast::cast_tree<ast::UnresolvedIdent>(methodDef.rhs);
+    auto expectedIvarName = methodDef.name.lookupWithAt(ctx);
+    return ivar != nullptr && ivar->kind == ast::UnresolvedIdent::Kind::Instance && expectedIvarName.exists() &&
+           ivar->name == expectedIvarName;
+}
```

Comment:
I thought this was pretty clever, as it requires a tiny amount of code, and trivially catches all current and future rewriter passes that emit things that look like `attr_reader` methods.

Reply:
Nice!